src/main/java/frc/robot/systems/GrabberFSM.java

```diff
@@ -0,0 +1,233 @@
+	 * as it may be called multiple times in a boot cycle,
+	 * Ex. if the robot is enabled, disabled, then reenabled.
+	 */
+	public void reset() {
+		currentState = FSMState.LOW_IDLE;
+
+		// Call one tick of update to ensure outputs reflect start state
+		update(null);
+	}
+	/**
+	 * Update FSM based on new inputs. This function only calls the FSM state
+	 * specific handlers.
+	 * @param input Global TeleopInput if robot in teleop mode or null if
+	 *        the robot is in autonomous mode.
+	 */
+	public void update(TeleopInput input) {
+		switch (currentState) {
+			case LOW_IDLE:
+				handleLowIdleState(input);
+				break;
+
+			case ASCENDING:
+				handleAscendingState(input);
+				break;
+            
+            case DESCENDING:
+                handleDescendingState(input);
+                break;
+            
+            case CONTROL:
+                handleControlState(input);
+                break;
+            
+            case MID_IDLE:
+                handleMidIdleState(input);
+                break;
+
+			default:
+				throw new IllegalStateException("Invalid state: " + currentState.toString());
+		}
+		currentState = nextState(input);
+	}
+
+	/* ======================== Private methods ======================== */
+	/**
+	 * Decide the next state to transition to. This is a function of the inputs
+	 * and the current state of this FSM. This method should not have any side
+	 * effects on outputs. In other words, this method should only read or get
+	 * values to decide what state to go to.
+	 * @param input Global TeleopInput if robot in teleop mode or null if
+	 *        the robot is in autonomous mode.
+	 * @return FSM state for the next iteration
+	 */
+	private FSMState nextState(TeleopInput input){
+        boolean isAscendingButtonPressed = joystick.getRawButton(0);
+        boolean isDescendingButtonPressed = joystick.getRawButton(1);
+        boolean isAtLowerThresh = motor.getEncoder().getPosition() < 0;
+        boolean isAtUpperThresh = motor.getEncoder().getPosition() > thresh;
+
+
+		switch (currentState) {
+			case LOW_IDLE:
+				if(input != null){
+                    if(isAscendingButtonPressed && !isDescendingButtonPressed){
+                        return FSMState.ASCENDING;
+                    } else{
+                        return FSMState.LOW_IDLE;
+                    }
+                }
+                return FSMState.LOW_IDLE;
+
+			case ASCENDING:
+				if(input != null){
+                    if(isAtUpperThresh){
+                        return FSMState.MID_IDLE;
+                    }else if(!isAscendingButtonPressed && isDescendingButtonPressed){
+                        return FSMState.DESCENDING;
+                    }else if(isAscendingButtonPressed && !isDescendingButtonPressed){
+                        return FSMState.ASCENDING;
+                    } else{
+                        return FSMState.CONTROL;
+                    }
+                }
+				return FSMState.LOW_IDLE;
```

Comment:
according to your FSM you shouldn't ever go back into LOW_IDLE. if input is null, then I think you should treat it as a "no buttons pressed" case